static/js/publisher/form/fileUpload.js

```diff
@@ -0,0 +1,102 @@
+import React from "react";
+
+import PropTypes from "prop-types";
+
+import { validateRestrictions } from "../../libs/fileValidation";
+
+class FileUpload extends React.Component {
```

Comment:
`FileUpload` name is a bit misleading, it doesn't seem to have anything to do with 'upload'. Maybe something like `FileValidator` or `FileValidationInput` would be more accurate?

Reply:
maybe just "File"? As it's essentially just a wrapper around a file input?

Reply:
`File` sound too much like native `File` class, so maybe `FileInput`?